src/vsync.js

```diff
@@ -51,12 +71,36 @@ export class Vsync {
    * @param {!Object<string, *>|undefined} opt_state
    */
   run(task, opt_state) {
+    // Do not request animation frames when the document is not visible.
+    if (!viewerFor(this.win).isVisible()) {
+      return;
```

Comment:
Should we consider assert in this case? Or at least an API that returns `false` if scheduling wasn't possible?